server/issue.go

```diff
@@ -1006,3 +1009,99 @@ func (s *Server) postInboxIssueActivity(ctx context.Context, issue *api.Issue, a
 
 	return nil
 }
+
+func (s *Server) getTenantDatabases(ctx context.Context, projectID int, databaseList []*api.Database, databaseName string) ([][]*api.Database, error) {
+	deployConfig, err := s.DeploymentConfigService.FindDeploymentConfig(ctx, &api.DeploymentConfigFind{
+		ProjectID: &projectID,
+	})
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch deployment config for project ID: %v", projectID)).SetInternal(err)
+	}
+	if deployConfig == nil {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Deployment config missing for project ID: %v", projectID)).SetInternal(err)
+	}
+	deploySchedule, err := api.ValidateAndGetDeploymentSchedule(deployConfig.Payload)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, "Failed to get deployment schedule").SetInternal(err)
+	}
+	for _, database := range databaseList {
+		if err := s.composeDatabaseRelationship(ctx, database); err != nil {
+			return nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to compose database relationship for database ID %v", database.ID)).SetInternal(err)
+		}
+	}
+	p, err := getPipelineFromDeploymentSchedule(deploySchedule, databaseName, databaseList)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, "Failed to create deployment pipeline").SetInternal(err)
+	}
+	return p, nil
+}
+
+func (s *Server) getTenantDatabaseSchema(ctx context.Context, instance *api.Instance, projectID int, databaseName string) (string, string, error) {
+	// Find all databases in the project.
+	databaseList, err := s.DatabaseService.FindDatabaseList(ctx, &api.DatabaseFind{
+		ProjectID: &projectID,
+	})
+	if err != nil {
+		return "", "", echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch databases in project ID: %v", projectID)).SetInternal(err)
+	}
+
+	pipeline, err := s.getTenantDatabases(ctx, projectID, databaseList, databaseName)
```

Comment:
ok, I guess pipeline and stage are unfortunate names because they refer to [][]database and []database. This is not good because we already have the pipeline and stage concept.

I would suggest to rename this pipeline/stage term to databaseMatrix/databaseList

Reply:
Not?